### DISCUSSION TITLE
Feature Request - Create a Dedicated Event for Changing Tile pages.  OnPageChanged()

### DISCUSSION THREAD

**Discussion by pt-fasaldivar** [elgatosf/elgato#37] [Ideas]
It would be great to have a dedicated event to that is triggered when the user changes tile pages on the StreamDeck device. I currently don't have a way of knowing when the user changes tile pages vs when he/she deletes a tile. Both actions seem to call the same WillDisappear event.  Please let me know if this is possible.

Thank you,

**Comment by markcellis** [comment]
I would also like to see this.

**Comment by graybilldustin** [comment]
Same

**Comment by ChekTek** [comment]
Hi @pt-fasaldivar 

We currently have a ticket internally for an event to let plugins know that a users has deleted a key. I'm hoping to get that prioritized soon!

 Do you have a specific use case for an event when the page changes, rather than utilizing the `willDisappear` event?

**Reply by graybilldustin** [reply]
When they delete a button we need to free up licensing in the connected software, when they change pages the license needs to stay consumed.

**Reply by ChekTek** [reply]
That makes sense, we definitely want to provide an event for when the action is deleted. If that were provided, do you still have a specific need for the page change event?

**Reply by graybilldustin** [reply]
I don't know if I do. When else does willdisappear get called. Isn't that called when you lock the pc as well? It might be nice to have a distinction between locking and changing pages but that's not a current requirement.

**Reply by ChekTek** [reply]
Awesome! Ok, I wanted to make sure I understand the main problem we are trying to solve. I've reported this for the team to be discussed and prioritized. Thanks!

**Reply by graybilldustin** [reply]
Any idea what the timeline would be for this? Thanks

**Reply by graybilldustin** [reply]
Good morning, any idea what the timeline might be for this? Thanks

**Reply by GeekyEggo** [reply]
Hi @graybilldustin, this functionality is not yet scheduled for an upcoming release at this stage. All of the latest release features, bug fixes, and improvements can be found on our [Maker Discord](https://discord.gg/GehBUcu627), more specifically our Maker Town Hall gatherings, whereby we discuss the roadmap for the SDK and Maker Tools.
